packages/expo-media-library/ios/EXMediaLibrary/EXMediaLibrary.m

```diff
@@ -422,9 +422,31 @@ - (NSDictionary *)constantsToExport
       [[PHImageManager defaultManager] requestAVAssetForVideo:asset
                                                       options:options
                                                 resultHandler:^(AVAsset * _Nullable asset, AVAudioMix * _Nullable audioMix, NSDictionary * _Nullable info) {
-        AVURLAsset *urlAsset = (AVURLAsset *)asset;
-        result[@"localUri"] = [[urlAsset URL] absoluteString];
-        resolve(result);
+        // Slow motion videos are returned as an AVComposition instance
+        if ([asset isKindOfClass:[AVComposition class]]) {
+            NSString *directory = [self.fileSystem.cachesDirectory stringByAppendingPathComponent:@"MediaLibrary"];
+            [self.fileSystem ensureDirExistsWithPath:directory];
+            NSString *videoOutoutFileName = [NSString stringWithFormat:@"slowMoVideo-%d.mov",arc4random() % 1000];
+            NSString *videoFileOutputPath = [directory stringByAppendingPathComponent:videoOutoutFileName];
+            NSURL *videoFileOutputURL = [NSURL fileURLWithPath:videoFileOutputPath];
+            
+            AVAssetExportSession *exporter = [[AVAssetExportSession alloc] initWithAsset:asset presetName:AVAssetExportPresetHighestQuality];
+            exporter.outputURL = videoFileOutputURL;
+            exporter.outputFileType = AVFileTypeQuickTimeMovie;
```

Comment:
Here, we are sure that the output file type is `QuickTimeMovie`, aren't we? I'm asking cause I'm not sure if the `AVComposition` can store only slow-motion videos.

Reply:
According to the definition on Apple’s documentation, AVComposition is meant for storing special “videos”, so yes I think exporting to QuickTimeMovie type is reasonable 🤔 https://developer.apple.com/documentation/avfoundation/avcomposition

Reply:
Special "videos" is a little bit ambiguous 😅 But exporting as `QuickTimeMovie` seems reasonable in this case ;)